src/Aspire.Hosting.Azure/AzureResourcePreparer.cs

```diff
@@ -353,6 +353,12 @@ private static void ProcessAzureReferences(HashSet<IAzureResource> azureReferenc
             return;
         }
 
+        if (value is IKeyVaultSecretReference keyVaultSecretReference)
+        {
+            azureReferences.Add(keyVaultSecretReference.KeyVaultResource);
```

Comment:
Naming consistency. Other resources have this property as `Resource` not `KeyVaultResource`.